tlsutil/config.go

```diff
@@ -1115,29 +1132,38 @@ func ParseCiphers(cipherStr string) ([]uint16, error) {
 	return suites, nil
 }
 
-// CipherString performs the inverse operation of ParseCiphers
-func CipherString(ciphers []uint16) (string, error) {
-	// Note: this needs to be kept up to date with the cipherMap in ParseCiphers
-	cipherMap := map[uint16]string{
-		tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA:    "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
-		tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256: "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
-		tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256: "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
-		tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA:    "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
-		tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384: "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
-		tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA:      "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
-		tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256:   "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
-		tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256:   "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
-		tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA:      "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
-		tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384:   "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+// ParseCiphers parse cipher suites from the comma-separated string into
+// recognized slice
+func ParseCiphers(cipherStr string) ([]types.TLSCipherSuite, error) {
+	cipherStr = strings.TrimSpace(cipherStr)
+	if cipherStr == "" {
+		return []types.TLSCipherSuite{}, nil
+	}
+	ciphers := strings.Split(cipherStr, ",")
+
+	suites := make([]types.TLSCipherSuite, len(ciphers))
+	for i, cipher := range ciphers {
+		suites[i] = types.TLSCipherSuite(cipher)
+	}
+
+	err := types.ValidateConsulAgentCipherSuites(suites)
+	if err != nil {
+		return []types.TLSCipherSuite{}, err
+	}
+
+	return suites, nil
+}
```

Comment:
Again, I wonder if this should live in the `config` package instead? Because the comma-separated string format is very much an "external" detail 🤷🏻